### PR TITLE
Modify initializer to check for valid paths at startup

### PR DESCRIPTION
**Pull request by nanaze**

### CODE REVIEW

**Comment by nanaze** [comment]
fixes #17

File: CONTRIBUTORS [hunk 1]
```diff
@@ -11,3 +11,4 @@
 
 Damien Martin-Guillerez <dmarting@google.com>
 Dmitry Lomov <dslomov@google.com>
+Nathan Naze <nanaze@gmail.com>
```
Comment (damienmg) [inline CONTRIBUTORS:14]: If you currently work at Google, please use your @google.com address

Add an empty line.

File: AUTHORS [hunk 2]
```diff
@@ -7,3 +7,4 @@
 # The email address is not required for organizations.
 
 Google Inc.
+Nathan Naze <nanaze@gmail.com>
```
Comment (damienmg) [inline AUTHORS:10]: If you are currently working at Google, remove your name, Google owns the copyright.

File: com.google.devtools.bazel.e4b/src/com/google/devtools/bazel/e4b/preferences/BazelPreferenceInitializer.java [hunk 3]
```diff
@@ -25,10 +28,24 @@
...
   @Override
   public void initializeDefaultPreferences() {
     IPreferenceStore store = Activator.getDefault().getPreferenceStore();
-    store.setDefault("BAZEL_PATH", Activator.DEFAULT_BAZEL_PATH);
+    store.setDefault("BAZEL_PATH", findBazelPath());
   }
 
+  private static String findBazelPath() {
+    for (String path : BAZEL_PATH_CANDIDATES) {
+      if (new File(path).isFile()) {
```
Comment (damienmg) [inline com.google.devtools.bazel.e4b/src/com/google/devtools/bazel/e4b/preferences/BazelPreferenceInitializer.java:42]: Adds test for executable flag?

File: com.google.devtools.bazel.e4b/src/com/google/devtools/bazel/e4b/preferences/BazelPreferenceInitializer.java [hunk 4]
```diff
@@ -25,10 +28,24 @@
...
+        return path;
+      }
+    }
+
+    System.err.println("No candidate path for blaze found. Using default.");
```
Comment (damienmg) [inline com.google.devtools.bazel.e4b/src/com/google/devtools/bazel/e4b/preferences/BazelPreferenceInitializer.java:47]: s/blaze/bazel

Comment (damienmg) [inline com.google.devtools.bazel.e4b/src/com/google/devtools/bazel/e4b/preferences/BazelPreferenceInitializer.java:47]: Do not use System.err.println, use Activator.error instead.

Comment (damienmg) [reply com.google.devtools.bazel.e4b/src/com/google/devtools/bazel/e4b/preferences/BazelPreferenceInitializer.java:47]: Also maybe tell in the message to go to the Eclipse preferences > Bazel to provide the path to Bazel.

**Review by damienmg — CHANGES_REQUESTED** [review]
Overall looks ok.

Ideally we would try to do a lookout on the path to find the bazel the user use, but this change is already a good improvement over the status quo.

**Comment by damienmg** [comment]
@nanaze friendly ping

**Comment by damienmg** [comment]
I have a concurrent change. Abandoning this one

**Pull request closed without merging**